src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java

```diff
@@ -1550,29 +1557,187 @@ public void testInvalidArrayInitWithTrueStrictCondition1()
+                "12, 31, 32"),
+            "44:15: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "array initialization", 14,
+                "12, 31, 32"),
+            "45:7: " + getCheckMessage(MSG_ERROR_MULTI, "array initialization rcurly", 6, "8, 12"),
+            "48:7: " + getCheckMessage(MSG_ERROR_MULTI, "array initialization lcurly", 6, "8, 12"),
+            "49:15: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "array initialization", 14,
+                "8, 10, 12"),
+            "51:7: " + getCheckMessage(MSG_ERROR_MULTI, "array initialization rcurly", 6, "8, 12"),
+            // following are tests for annotation array initialization
+            "59:13: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "annotation array initialization",
+                12, "16, 46, 48"),
+            "67:15: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "annotation array initialization",
+                14, "16, 28, 30"),
+            "68:9: " + getCheckMessage(MSG_ERROR_MULTI, "annotation array initialization rcurly",
+                8, "12, 16"),
+            "70:13: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "annotation array initialization",
+                12, "16, 29, 31"),
         };
 
         // Test input for this test case is not checked due to issue #693.
+        verifyWarns(checkConfig, fileName, expected);
+    }
+
+    // Test input due to discussion #16906
+    @Test
+    public void testInvalidArrayInitIndentNoCommentsTrueStrictCondition()
+            throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
+
+        checkConfig.addProperty("arrayInitIndent", "4");
+        checkConfig.addProperty("basicOffset", "4");
+        checkConfig.addProperty("braceAdjustment", "0");
+        checkConfig.addProperty("caseIndent", "4");
+        checkConfig.addProperty("forceStrictCondition", "true");
+        checkConfig.addProperty("lineWrappingIndentation", "4");
+        checkConfig.addProperty("tabWidth", "4");
+        checkConfig.addProperty("throwsIndent", "4");
+        final String fileName = getPath(
+            "InputIndentationInvalidArrayInitIndentWithoutTrailingComments.java");
+        final String[] expected = {
+            "28:13: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "annotation array initialization",
+                12, "16, 46, 48"),
+            "32:15: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "annotation array initialization",
+                14, "12, 16"),
+            "36:15: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "annotation array initialization",
+                14, "16, 28, 30"),
+            "37:9: " + getCheckMessage(MSG_ERROR_MULTI, "annotation array initialization rcurly",
+                8, "12, 16"),
+            "39:13: " + getCheckMessage(MSG_CHILD_ERROR, "annotation array initialization",
+                12, 16),
+        };
+        verify(checkConfig, fileName, expected);
+    }
+
+    // Test input due to discussion #16906
+    @Test
```

Comment:
`Test Input without trailing comment and usage of 'verify' method is due to #16906`